app/layersproxymodel.cpp

```diff
@@ -153,3 +153,17 @@ QVariant LayersProxyModel::getData( QModelIndex index, int role ) const
 {
   return sourceModel()->data( index, role );
 }
+
+bool LayersProxyModel::isPositionTrackingLayer( QgsMapLayer *layer ) const
+{
+  if ( !layer )
+    return false;
+
+  QgsProject *project = QgsProject::instance();
```

Comment:
Everything looks good, the only bit is that we should not read the QGIS project instance as singleton. Let's pass it in as parameter instead (via QML?)